### PR TITLE
Test that ::backdrop receives events for the associated element

### PR DESCRIPTION
**Pull request by nt1m**

### CODE REVIEW

File: html/semantics/interactive-elements/the-dialog-element/backdrop-receives-element-events.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-actions.js"></script>
+<script>
+const dialog = document.querySelector("dialog");
+dialog.showModal();
+async_test(function() {
+    dialog.addEventListener("click", () => {
+        dialog.className = "clicked";
+        this.done();
```
Comment (foolip) [inline html/semantics/interactive-elements/the-dialog-element/backdrop-receives-element-events.html:37]: This is what makes the test pass, not the red/green color change. Can you add a comment earlier that the color is just for debugging, and has no effect on the test?

File: html/semantics/interactive-elements/the-dialog-element/backdrop-receives-element-events.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+<script src="/resources/testdriver-vendor.js"></script>
+<script>
+const dialog = document.querySelector("dialog");
+dialog.showModal();
+async_test(function() {
+    dialog.addEventListener("click", () => {
+        dialog.className = "clicked";
+        this.done();
+    });
+    new test_driver.Actions()
```
Comment (foolip) [inline html/semantics/interactive-elements/the-dialog-element/backdrop-receives-element-events.html:39]: This returns a promise which could be rejected. Can you either handle that and fail the test (using `this.unreached_func('description')` or similar, or alternatively convert this to `promise_test` and turn the event into a promise instead? (Yes, tests that mix events and promises are pretty annoying.)

Or, as an option to simplify, you could use `setup({ single_test: true })`, that way a rejected promise will fail the single test without any glue code.

Comment (nt1m) [reply html/semantics/interactive-elements/the-dialog-element/backdrop-receives-element-events.html]: Used the single_test setup, thanks!

File: html/semantics/interactive-elements/the-dialog-element/backdrop-receives-element-events.html (hunk 1 above)
Comment (nt1m) [reply html/semantics/interactive-elements/the-dialog-element/backdrop-receives-element-events.html]: Done!

**Review by foolip — APPROVED** [review]
Test looks good to me, but are you having trouble making it pass CI?

**Pull request merged**